### PR TITLE
Adapt to core changes (ThingHandlerService)

### PR DESCRIPTION
**Pull request by J-N-K**
See https://github.com/openhab/openhab-core/pull/3957

### CODE REVIEW

**Review by bern77 — APPROVED** [review]
LGTM for HeliosEasyControls

**Review by clinique — APPROVED** [review]

**Comment by J-N-K** [comment]
@openhab/add-ons-maintainers Please make sure that every contribution of thing-bound actions/providers/services follows the new coding scheme. Since this is a breaking change, older and newer versions that make use of `ThingHandlerService` will no longer work with the old code.

**Review by EvilPingu — APPROVED** [review]

**Review by druciak — APPROVED** [review]

File: bundles/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/internal/discovery/VerisureThingDiscoveryService.java
Comment (wborn) [inline bundles/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/internal/discovery/VerisureThingDiscoveryService.java:107]: I think it should also call `super.initialize();` here?

File: bundles/org.openhab.binding.vesync/src/main/java/org/openhab/binding/vesync/internal/discovery/VeSyncDiscoveryService.java
Comment (wborn) [inline bundles/org.openhab.binding.vesync/src/main/java/org/openhab/binding/vesync/internal/discovery/VeSyncDiscoveryService.java:73]: I think it should also call `super.initialize();` here?

File: bundles/org.openhab.binding.windcentrale/src/main/java/org/openhab/binding/windcentrale/internal/WindcentraleDiscoveryService.java
Comment (wborn) [inline bundles/org.openhab.binding.windcentrale/src/main/java/org/openhab/binding/windcentrale/internal/WindcentraleDiscoveryService.java:72]: I think it should also call `super.initialize();` here?

File: bundles/org.openhab.binding.wled/src/main/java/org/openhab/binding/wled/internal/WLedSegmentDiscoveryService.java
Comment (wborn) [inline bundles/org.openhab.binding.wled/src/main/java/org/openhab/binding/wled/internal/WLedSegmentDiscoveryService.java:85]: I think it should also call `super.initialize();` here?

**Review by wborn — COMMENTED** [review]
Thanks for your huge work! It nicely cleans up all the workarounds. :+1:

I only found these comments:

**Review by david-pace — APPROVED** [review]
LGTM for `boschshc`

**Comment by J-N-K** [comment]
done

**Review by wborn — APPROVED** [review]
Thank you very much!

**Review by mvalla — APPROVED** [review]
LGTM for OpenWebNet

File: bundles/org.openhab.binding.pilight/src/main/java/org/openhab/binding/pilight/internal/discovery/PilightDeviceDiscoveryService.java
Comment (stefanroellin) [inline bundles/org.openhab.binding.pilight/src/main/java/org/openhab/binding/pilight/internal/discovery/PilightDeviceDiscoveryService.java:184]: Should super.dispose() be called here?

**Review by stefanroellin — COMMENTED** [review]
If I read the annotation in AbstractThingHandlerDiscoveryService correctly, we can assume that thingHandler is not null and it is not necessary to check for null. Is this correct? If so, some checks in PilightDeviceDiscoveryService can be removed. I could do this in a follow up pull request after this is merged.

File: bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/discovery/AVMFritzDiscoveryService.java
Comment (wborn) [inline bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/discovery/AVMFritzDiscoveryService.java:61]: I think it should also call `super.initialize();` here?

File: bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/discovery/AVMFritzDiscoveryService.java
Comment (wborn) [inline bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/discovery/AVMFritzDiscoveryService.java:66]: I think it should also call `super.dispose();` here?

File: bundles/org.openhab.binding.asuswrt/src/main/java/org/openhab/binding/asuswrt/internal/AsuswrtDiscoveryService.java
Comment (wborn) [inline bundles/org.openhab.binding.asuswrt/src/main/java/org/openhab/binding/asuswrt/internal/AsuswrtDiscoveryService.java:56]: I think it should also call `super.dispose();` here?

File: bundles/org.openhab.binding.boschindego/src/main/java/org/openhab/binding/boschindego/internal/discovery/IndegoDiscoveryService.java
Comment (wborn) [inline bundles/org.openhab.binding.boschindego/src/main/java/org/openhab/binding/boschindego/internal/discovery/IndegoDiscoveryService.java:85]: I think it should also call `super.dispose();` here?

**Review by wborn — COMMENTED** [review]
There are a few more overridden methods that have this issue:

File: bundles/org.openhab.binding.pilight/src/main/java/org/openhab/binding/pilight/internal/discovery/PilightDeviceDiscoveryService.java
Comment (J-N-K) [reply bundles/org.openhab.binding.pilight/src/main/java/org/openhab/binding/pilight/internal/discovery/PilightDeviceDiscoveryService.java:184]: correct.

**Comment by J-N-K** [comment]
Done.

**Review by stefanroellin — APPROVED** [review]
LGTM for pilight. Thanks for the changes.

**Review by MrRonfo — APPROVED** [review]
LGTM for bticinosmarther

File: bundles/org.openhab.binding.qolsysiq/src/main/java/org/openhab/binding/qolsysiq/internal/discovery/QolsysIQChildDiscoveryService.java
Comment (wborn) [inline bundles/org.openhab.binding.qolsysiq/src/main/java/org/openhab/binding/qolsysiq/internal/discovery/QolsysIQChildDiscoveryService.java:51]: I think it should also call `super.initialize();` here?

File: bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/TapoDiscoveryService.java
Comment (wborn) [inline bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/TapoDiscoveryService.java:69]: I think it should also call `super.initialize();` here?

File: bundles/org.openhab.binding.tr064/src/main/java/org/openhab/binding/tr064/internal/Tr064DiscoveryService.java
Comment (wborn) [inline bundles/org.openhab.binding.tr064/src/main/java/org/openhab/binding/tr064/internal/Tr064DiscoveryService.java:54]: I think it should also call `super.dispose();` here?

**Review by wborn — COMMENTED** [review]
Looks like there are a few more:

File: bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/TapoDiscoveryService.java
Comment (J-N-K) [reply bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/TapoDiscoveryService.java:69]: Debatable, because the service handles its background discovery by itself and the `activate` method didn't call `super.activate` either. I have refactored that to use the standard mechanism, unfortunately I can't test it.

**Comment by J-N-K** [comment]
done

**Review by gdolfen — APPROVED** [review]
LGTO for enigma2

**Review by wborn — APPROVED** [review]

**Pull request merged**
